lang/java/avro/src/main/java/org/apache/avro/specific/SpecificData.java

```diff
@@ -122,6 +122,11 @@ public DatumWriter createDatumWriter(Schema schema) {
   /** Return the singleton instance. */
   public static SpecificData get() { return INSTANCE; }
 
+  private boolean useCustomCoderFlag
+    = Boolean.parseBoolean(System.getProperty("org.apache.avro.specific.use_custom_coders","false"));
+  public boolean useCustomCoders() { return useCustomCoderFlag; }
+  public void setCustomCoders(boolean flag) { useCustomCoderFlag = flag; }
+
```

Comment:
These new public methods need javadoc.

Reply:
Regarding compiler options: the controlability here isn't intended to be a long-term option, it's intended to be a "feature flag" that allows more risk-seeking users to use the feature early and help find corner-case bugs.  After a release or two, the flag should go away and this should be the only way to generate code.  As a result, this feature flag is more appropriate as a runtime option (as it is now), vs. a compile-time option.  For example, imagine the scenario where the custom coders look really good during testing, they get shipped to production, and a bug is tickled.  In this scenario, the ops team will want to turn off the custom coders with a runtime flag, rather than wait for the system to be recompiled with a change to the compile-time flag.

Reply:
Regarding documentation: there is a larger issue here that this feature flag isn't documented.  I'll add the suggested Javadoc but also look for other places to document this new feature and how Avro users can enable it.

Reply:
Regarding piecemeal: yes, it would definitely be valuable to have a finite list of todo's here rather than an indefinitely growing one :-)